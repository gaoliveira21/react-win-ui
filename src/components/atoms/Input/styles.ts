import styled, { css } from 'styled-components'

export const Input = styled.input`
  ${({ theme }) => css`
    background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};
    border: 1px solid ${theme.colors.elevation.textControl.BORDER};
    border-bottom: 2px solid ${theme.colors.stroke.control.SECONDARY};

    color: ${theme.colors.text.PRIMARY};
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};

    padding: 4px 35px 6px 11px;
    border-radius: 3px;

    &::placeholder {
      font-size: ${theme.fonts.text.BODY.size};
      font-weight: ${theme.fonts.text.BODY.weight};
      line-height: ${theme.fonts.text.BODY.lineHeight};
      color: ${theme.colors.text.SECONDARY};
    }

    &:focus {
      outline: none;
      border-bottom: 2px solid ${theme.colors.fill.accent.ACCENT_PRIMARY};
      background-color: ${theme.colors.fill.control.ACTIVE};
    }

    &:hover {
      &:not(:focus, :disabled) {
        background-color: ${theme.colors.fill.control.CONTROL_SECONDARY};
      }
    }

    &:disabled {
      background-color: ${theme.colors.fill.control.DISABLED};
      color: ${theme.colors.text.DISABLED};
      cursor: not-allowed;
    }
  `}
`
