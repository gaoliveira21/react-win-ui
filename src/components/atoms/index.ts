export * from './Text'
export * from './Heading'
export * from './Button'
export * from './Input'
export * from './Label'
