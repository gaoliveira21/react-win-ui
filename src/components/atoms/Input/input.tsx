import { forwardRef } from 'react'

import * as S from './styles'

export type InputProps = React.InputHTMLAttributes<HTMLInputElement>

export const Input = forwardRef<HTMLInputElement, InputProps>(
  (props, ref): React.ReactElement => {
    return <S.Input ref={ref} {...props} />
  }
)
