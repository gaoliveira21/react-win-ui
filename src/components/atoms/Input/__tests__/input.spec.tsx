import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Input } from '@/components/atoms/Input'
import { renderWithTheme } from '@/tests/render-with-theme'

describe('<Input />', () => {
  it('Should calls onChange when user types', () => {
    const onChange = jest.fn()
    renderWithTheme(<Input placeholder="input" onChange={onChange} />)

    const input = screen.getByPlaceholderText(/input/i)
    const typedValue = 'any_text'
    userEvent.type(input, typedValue)

    expect(onChange).toHaveBeenCalled()

    const event = onChange.mock.calls[0][0]

    expect(event.target.value).toBe(typedValue)
    expect(input).toMatchSnapshot()
  })

  it('Should receive a value prop', () => {
    const value = 'any_value'
    renderWithTheme(<Input placeholder="input" value={value} readOnly />)

    expect(screen.getByPlaceholderText(/input/i)).toHaveValue(value)
  })
})
