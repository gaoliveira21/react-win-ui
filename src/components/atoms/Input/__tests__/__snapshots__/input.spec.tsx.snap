// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should calls onChange when user types 1`] = `
.c0 {
  background-color: rgba(255,255,255,0.7);
  border: 1px solid rgba(0,0,0,0.05);
  border-bottom: 2px solid rgba(0,0,0,0.22);
  color: rgb(0,0,0);
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  padding: 4px 35px 6px 11px;
  border-radius: 3px;
}

.c0::-webkit-input-placeholder {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  color: rgba(0,0,0,0.63);
}

.c0::-moz-placeholder {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  color: rgba(0,0,0,0.63);
}

.c0:-ms-input-placeholder {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  color: rgba(0,0,0,0.63);
}

.c0::placeholder {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  color: rgba(0,0,0,0.63);
}

.c0:focus {
  outline: none;
  border-bottom: 2px solid rgb(0,95,184);
  background-color: rgb(255,255,255);
}

.c0:hover:not(:focus,:disabled) {
  background-color: rgba(249,249,249,0.5);
}

.c0:disabled {
  background-color: rgba(249,249,249,0.3);
  color: rgba(0,0,0,0.36);
  cursor: not-allowed;
}

<input
  class="c0"
  placeholder="input"
/>
`;
